From 7dc0d29a3ef6e45455f537c8187791446ca7e1e5 Mon Sep 17 00:00:00 2001
From: Rok Mandeljc <rok.mandeljc@gmail.com>
Date: Sun, 26 Mar 2017 17:09:42 +0200
Subject: VHBA: compatibility fix for upcoming kernel 4.11

As per Bug #104, from 4.11-rc2 on, we need to include linux/sched/signal.h
instead of linux/sched.h

diff --git a/vhba-module/vhba.c b/vhba-module/vhba.c
index 4a14a10e..ff30e4cb 100644
--- a/vhba-module/vhba.c
+++ b/vhba-module/vhba.c
@@ -18,16 +18,21 @@
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
 
+#include <linux/version.h>
+
 #include <linux/init.h>
 #include <linux/module.h>
 #include <linux/highmem.h>
 #include <linux/fs.h>
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 11, 0)
+#include <linux/sched/signal.h>
+#else
 #include <linux/sched.h>
+#endif
 #include <linux/platform_device.h>
 #include <linux/miscdevice.h>
 #include <linux/poll.h>
 #include <linux/slab.h>
-#include <linux/version.h>
 #ifdef CONFIG_COMPAT
 #include <linux/compat.h>
 #endif
